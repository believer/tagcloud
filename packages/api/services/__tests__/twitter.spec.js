const { getToken, getTweets } = require('../twitter')
const got = require('got')
const mockdate = require('mockdate')

jest.mock('got')
jest.mock('randomcolor', () => jest.fn().mockReturnValue('#0088ff'))
jest.mock('form-data', () =>
  jest.fn(() => ({
    append: jest.fn(),
  }))
)

describe('#getToken', () => {
  test('should get a token', async () => {
    got.post.mockResolvedValueOnce({
      body: '{"access_token":"token"}',
    })

    await getToken()

    expect(got.post.mock.calls[0]).toMatchSnapshot()
  })

  test('should return token', async () => {
    got.post.mockResolvedValueOnce({
      body: '{"access_token":"token"}',
    })

    await expect(getToken()).resolves.toMatchSnapshot()
  })
})

describe('#getTweets', () => {
  test('should get a token and get tweets', async () => {
    mockdate.set('2019-04-01')

    got.post
      .mockResolvedValueOnce({
        body: '{"access_token":"token"}',
      })
      .mockResolvedValueOnce({
        body: {
          results: [],
        },
      })

    await getTweets('testtag')

    // Only validate second call because previous test asserts on first call for token
    expect(got.post.mock.calls[1]).toMatchSnapshot()
  })

  test('should return a list of the 100 most used words', async () => {
    mockdate.set('2019-04-01')

    got.post
      .mockResolvedValueOnce({
        body: '{"access_token":"token"}',
      })
      .mockResolvedValueOnce({
        body: {
          results: [
            {
              text:
                '"Amet" √∂kar cum vel #nobis iste nemo: Aut voluptas ut saepe tempora similique. Illo facilis at tempora fuga nostrum? Animi molestiae consectetur aliquam at explicabo Reprehenderit architecto & &amp; consequatur enim commodi & deleniti',
            },
            {
              text:
                'ConseüòÖctetur iste √•ter alias\n\n dolores atque tenetur Numquam ipsa tempore ad dolore aspernatur Unde laudantium possimus temporibus iste illum libero laborum! Quas nemo exercitationem quibusdam amet ipsam perspiciatis Molestiae unde & laborum. √§ta',
            },
            {
              text:
                'Elit‚Ä¶ sit ullam https//tco/siqurycll7 optio √•ter sit perspiciatis. Et quos esse nihil officia enim Consequatur pariatur quo commodi adipisci voluptates dicta In aut non molestiae corporis consequuntur! Alias ex nihil rem &amp; at',
            },
            {
              text:
                'AmüòÅet! eaque... ¬øamet https//tco/siqurycll7 reprehenderit, labore a! Explicabo fugiat odit mollitia saepe labore tempore, doloremque. Consectetur quod rem quidem consectetur eaque. Perspiciatis est @amet eligendi ut labore sapiente excepturi. √§ta Perferendis &amp; √∂kar natus!',
            },
          ],
        },
      })

    const result = await getTweets('testtag')

    expect(result).toMatchSnapshot()
    expect(result.length).toBeLessThanOrEqual(100)
  })
})
