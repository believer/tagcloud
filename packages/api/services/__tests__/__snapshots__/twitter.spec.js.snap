// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getToken should get a token 1`] = `
Array [
  "https://api.twitter.com/oauth2/token",
  Object {
    "auth": "undefined:undefined",
    "body": Object {
      "append": [MockFunction] {
        "calls": Array [
          Array [
            "grant_type",
            "client_credentials",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
]
`;

exports[`#getToken should return token 1`] = `"token"`;

exports[`#getTweets should get a token and get tweets 1`] = `
Array [
  "https://api.twitter.com/oauth2/token",
  Object {
    "auth": "undefined:undefined",
    "body": Object {
      "append": [MockFunction] {
        "calls": Array [
          Array [
            "grant_type",
            "client_credentials",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
]
`;

exports[`#getTweets should return a list of the 100 most used words 1`] = `
Array [
  Object {
    "count": 5,
    "style": Object {
      "color": "hsl(0, 0%, 0%)",
      "fontSize": "61.00px",
    },
    "word": "amet",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "cum",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "vel",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "nobis",
  },
  Object {
    "count": 3,
    "style": Object {
      "color": "hsl(0, 0%, 20%)",
      "fontSize": "41.80px",
    },
    "word": "iste",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "nemo",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "aut",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "voluptas",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "saepe",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "tempora",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "similique",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "illo",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "facilis",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "fuga",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "nostrum?",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "animi",
  },
  Object {
    "count": 3,
    "style": Object {
      "color": "hsl(0, 0%, 20%)",
      "fontSize": "41.80px",
    },
    "word": "molestiae",
  },
  Object {
    "count": 4,
    "style": Object {
      "color": "hsl(0, 0%, 10%)",
      "fontSize": "51.40px",
    },
    "word": "consectetur",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "aliquam",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "explicabo",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "reprehenderit",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "architecto",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "consequatur",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "enim",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "commodi",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "deleniti",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "alias",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "dolores",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "atque",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "tenetur",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "numquam",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "ipsa",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "tempore",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "dolore",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "aspernatur",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "unde",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "laudantium",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "possimus",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "temporibus",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "illum",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "libero",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "laborum",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quas",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "exercitationem",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quibusdam",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "ipsam",
  },
  Object {
    "count": 3,
    "style": Object {
      "color": "hsl(0, 0%, 20%)",
      "fontSize": "41.80px",
    },
    "word": "perspiciatis",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "elit",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "sit",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "ullam",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "optio",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quos",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "esse",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "nihil",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "officia",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "pariatur",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quo",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "adipisci",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "voluptates",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "dicta",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "non",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "corporis",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "consequuntur",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "rem",
  },
  Object {
    "count": 2,
    "style": Object {
      "color": "hsl(0, 0%, 30%)",
      "fontSize": "32.20px",
    },
    "word": "eaque",
  },
  Object {
    "count": 3,
    "style": Object {
      "color": "hsl(0, 0%, 20%)",
      "fontSize": "41.80px",
    },
    "word": "labore",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "fugiat",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "odit",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "mollitia",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "doloremque",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quod",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "quidem",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "est",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "eligendi",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "sapiente",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "excepturi",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "perferendis",
  },
  Object {
    "count": 1,
    "style": Object {
      "color": "hsl(0, 0%, 40%)",
      "fontSize": "22.60px",
    },
    "word": "natus",
  },
]
`;
